cmd/server/server.go

```diff
@@ -43,12 +47,31 @@ func serverCmd(errorf errorFn) *cobra.Command {
 		Use:   "server",
 		Short: "Starts the mixer as a server",
 		Run: func(cmd *cobra.Command, args []string) {
-			fmt.Printf("Starting gRPC server on port %v", sa.port)
+			fmt.Printf("Starting gRPC server on port %v, type 'exit' to stop gracefully.\n", sa.port)
 
-			err := runServer(sa)
+			// The tracer will keep track of spans in memory and print them to stdout when the server finishes.
+			recorder := tracer.NewInMemoryRecorder()
+			t := tracer.New(recorder)
+			defer func() {
+				for _, span := range recorder.GetSpans() {
+					fmt.Printf("%v\n", span)
+				}
+			}()
+
+			srv, err := newServer(sa, t)
 			if err != nil {
 				errorf("%v", err)
+				os.Exit(1)
```

Comment:
errorf calls os.Exit, so here and elsewhere don't call it elsewhere. The idea is that in a test environment, the test can supply a different errorf function instead of having the code under test call os.Exit.